nebula-logger/core/main/logger-engine/lwc/logger/logEntryBuilder.js

```diff
@@ -107,28 +95,40 @@ const LogEntryBuilder = class {
     }
 
     _logToConsole() {
-        if (!this.isConsoleLoggingEnabled) {
-            return;
-        }
+        this.#settingsPromise().then(setting => {
+            this.isConsoleLoggingEnabled = setting.isConsoleLoggingEnabled;
 
-        /* eslint-disable no-console */
-        switch (this.loggingLevel) {
-            case 'ERROR':
-                console.error(this.message, this);
-                break;
-            case 'WARN':
-                console.warn(this.message, this);
-                break;
-            case 'INFO':
-                console.info(this.message, this);
-                break;
-            default:
-                console.debug(this.message, this);
-                break;
-        }
+            if (!this.isConsoleLoggingEnabled) {
+                return;
+            }
+
+            const consoleMessagePrefix = '%c Nebula Logger ';
+            const consoleFormatting = 'background: #0c598d; color: #fff; font-size: 12px; font-weight:bold;';
+            let consoleLoggingFunction;
+            switch (this.loggingLevel) {
+                case 'ERROR':
+                    consoleLoggingFunction = console.error;
+                    break;
+                case 'WARN':
+                    consoleLoggingFunction = console.warn;
+                    break;
+                case 'INFO':
+                    consoleLoggingFunction = console.info;
+                    break;
+                default:
+                    consoleLoggingFunction = console.debug;
+                    break;
+            }
+
+            consoleLoggingFunction(consoleMessagePrefix, consoleFormatting, this.message);
+            console.groupCollapsed(consoleMessagePrefix, consoleFormatting, 'Details for: ' + this.message);
+            consoleLoggingFunction(JSON.parse(JSON.stringify(this)));
+            console.trace();
+            console.groupEnd();
+        });
```

Comment:
This is pretty neat, it is scope creep, but it makes sense.

Reply:
Yeah, I didn't want to scope-creep too much, but this was something I've wanted to do for a while to help make the console output look nicer & easier to read